### PR TITLE
Basic Sigil implementation

### PR DESCRIPTION
**Pull request by Arcaratus**
-Added the ItemSigilToggleable to make making toggleable sigils easier
-Added the Green Grove Sigil, Blood Light Sigil, Fast Miner Sigil, Elemental Affinity Sigil, Magnetism Sigil, and Suppression Sigil
-Added necessary blocks and entities for Blood Light and Suppression Sigils
-Fixed Water and Lava Sigils taking liquid blocks as placeable blocks

### CODE REVIEW

File: src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilBase.java [hunk 1]
```diff
@@ -53,6 +54,11 @@ public boolean onItemUse(ItemStack stack, EntityPlayer player, World world, Bloc
     }
 
     @Override
+    public void onUpdate(ItemStack stack, World world, Entity entityIn, int itemSlot, boolean isSelected) {
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilBase.java]: Why override this..?

File: src/main/java/WayofTime/bloodmagic/registry/ModBlocks.java [hunk 2]
```diff
@@ -7,6 +7,9 @@
 import WayofTime.bloodmagic.BloodMagic;
 import WayofTime.bloodmagic.ConfigHandler;
 import WayofTime.bloodmagic.api.Constants;
+import WayofTime.bloodmagic.block.*;
+import WayofTime.bloodmagic.item.block.*;
+import WayofTime.bloodmagic.tile.*;
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/registry/ModBlocks.java]: If you're going to use these wildcards, you can cleanup the rest

File: src/main/java/WayofTime/bloodmagic/tile/TileSpectralBlock.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.ITickable;
+import net.minecraft.world.World;
+
+public class TileSpectralBlock extends TileEntity implements ITickable {
+
+    private int ticksRemaining;
+    private int containedBlockID;
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/tile/TileSpectralBlock.java]: No. Do not use numeric ID's.

**Comment by TehNut** [comment]
Check those 3 things and squash. Then I'll merge

File: src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilBase.java [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import WayofTime.bloodmagic.item.ItemBindable;
 import WayofTime.bloodmagic.util.helper.TextHelper;
 import lombok.Getter;
+import net.minecraft.entity.Entity;
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilBase.java]: Unnecessary import

**Comment by WayofTime** [comment]
https://github.com/WayofTime/BloodMagic/pull/515/files#diff-2e6a04868f9eb96dc328b1e1f0ff775aR99
With here, you can merge all of the if statements together, since you can do pos.offset(EnumFacing) or similar.

**Pull request closed without merging**